code/modules/projectiles/projectile.dm

```diff
@@ -13,7 +13,7 @@
 	randpixel = 0	
 	var/bumped = 0		//Prevents it from hitting more than one guy at once
 	var/def_zone = ""	//Aiming at
-	var/mob/firer = null//Who shot it
+	var/atom/movable/firer = null//Who shot it
```

Comment:
If you feel like it, and if you don't find weakrefs horrible, changing this to a weakref would be good for the same reasons as `target` earlier.